src/domain/relay/limit-addresses.mapper.ts

```diff
@@ -2,66 +2,110 @@ import { Inject, Injectable } from '@nestjs/common';
 import { Hex } from 'viem/types/misc';
 import { Erc20ContractHelper } from '@/domain/relay/contracts/erc20-contract.helper';
 import { SafeContractHelper } from '@/domain/relay/contracts/safe-contract.helper';
-import { ILoggingService, LoggingService } from '@/logging/logging.interface';
+import { ISafeRepository } from '@/domain/safe/safe.repository.interface';
 import { MultiSendDecoder } from '@/domain/contracts/contracts/multi-send-decoder.helper';
 import { ProxyFactoryDecoder } from '@/domain/relay/contracts/proxy-factory-decoder.helper';
 import {
   getSafeSingletonDeployment,
   getSafeL2SingletonDeployment,
+  getMultiSendCallOnlyDeployment,
 } from '@safe-global/safe-deployments';
 import { SafeDecoder } from '@/domain/contracts/contracts/safe-decoder.helper';
-import { isHex } from 'viem';
+import { isAddress, isHex } from 'viem';
 
 @Injectable()
 export class LimitAddressesMapper {
   // TODO: Abstract with that from SafeContractHelper
   private static SUPPORTED_SAFE_VERSION = '1.3.0';
 
   constructor(
-    @Inject(LoggingService) private readonly loggingService: ILoggingService,
+    @Inject(ISafeRepository)
+    private readonly safeRepository: ISafeRepository,
     private readonly safeContract: SafeContractHelper,
     private readonly erc20Contract: Erc20ContractHelper,
     private readonly safeDecoder: SafeDecoder,
     private readonly multiSendDecoder: MultiSendDecoder,
     private readonly proxyFactoryDecoder: ProxyFactoryDecoder,
   ) {}
 
-  getLimitAddresses(relayPayload: {
+  async getLimitAddresses(args: {
     chainId: string;
     to: string;
     data: string;
-  }): readonly Hex[] {
-    if (!isHex(relayPayload.to) || !isHex(relayPayload.data)) {
+  }): Promise<readonly Hex[]> {
+    // Ensure that the recipient and calldata are valid
+    if (!isAddress(args.to) || !isHex(args.data)) {
```

Comment:
Not strictly related to this PR, but does it make sense to have separate checks/error messages for those conditions? (`to` and `data` validity)

Reply:
I separated the error messages for each in ca9b11ca52cb5aabda07baca59bb1391fb567ec6.